osu.Game/Beatmaps/BeatmapUpdater.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Game.Database;
+using osu.Game.Rulesets.Objects;
+using Realms;
+
+namespace osu.Game.Beatmaps
+{
+    /// <summary>
+    /// Handles all processing required to ensure a local beatmap is in a consistent state with any changes.
+    /// </summary>
+    public class BeatmapUpdater
+    {
+        private readonly IWorkingBeatmapCache workingBeatmapCache;
+        private readonly BeatmapOnlineLookupQueue onlineLookupQueue;
+        private readonly BeatmapDifficultyCache difficultyCache;
+
+        public BeatmapUpdater(IWorkingBeatmapCache workingBeatmapCache, BeatmapOnlineLookupQueue onlineLookupQueue, BeatmapDifficultyCache difficultyCache)
+        {
+            this.workingBeatmapCache = workingBeatmapCache;
+            this.onlineLookupQueue = onlineLookupQueue;
+            this.difficultyCache = difficultyCache;
+        }
+
+        /// <summary>
+        /// Queue a beatmap for background processing.
+        /// </summary>
+        public void Queue(Live<BeatmapSetInfo> beatmap)
+        {
+            // For now, just fire off a task.
+            // TODO: Add actual queueing probably.
+            Task.Factory.StartNew(() => beatmap.PerformRead(Process));
```

Comment:
This is intended to be called from other sources, such as a server event which suggests a beatmap has been updated. I'm not sure whether this should be split out into smaller pieces (ie. in the case of an online update difficulty etc. would not need to be updated), potentially a `[Flags]` enum with the pieces that require update.

Can remove for now if preferred.